js/service.js

```diff
@@ -29,13 +29,14 @@ function doXHR(route, params = {}) {
   return new Promise((resolve, reject) => {
     request.open(`GET`, `${route}${combinedParams ? pojoToQuery(combinedParams) : ``}`, true);
 
+    request.withCredentials = true;
     request.onload = (event) => {
       let result = event.currentTarget;
 
       if (result.status >= 200 && result.status < 400) {
-        resolve(JSON.parse(result.response));
+        resolve(result.response);
```

Comment:
@Pomax in some cases we get redirected to the root route on Pulse API. It's status `200` but the response is just the html of the server's home page.

This is what I intend to change the above code to:

```js
if (result.status >= 200 && result.status < 400) {
  try {
    resolve(JSON.parse(result.response));
  } catch (error) {
    resolve(error);
  }
} else {
  reject(`XHR request failed, status ${result.status}.`);
}
```